### PR TITLE
refactor behave steps

### PR DESCRIPTION
**Pull request by charbeljc**
* add a 'strict' option to csv import to check that all data are loaded
* give access to imported objects in the following steps

### CODE REVIEW

**Comment by guewen** [comment]
Thanks, this is a good step forward.
Most of the code of the 2 functions `impl_import_cvs_with_delimiter` and `impl_import_cvs_with_options` is still redundant. Could you push it further and have only one function with arguments `ctx, model_name, csvfile, sep=",", options=None`, the steps only forwarding to it with the appropriate args?

File: features/steps/tools.py [hunk 1]
```diff
@@ -74,34 +97,57 @@ def impl(ctx, model_name, csvfile):
     path = os.path.join(*tmp_path)
     assert os.path.exists(path)
     sep = options.get('delimiter', ',')
+    strict = str2bool(options.get('strict', 'false'))
     data = csv.reader(open(path, 'rb'), delimiter=str(sep))
     head = data.next()
     # generator does not work
     values = [x for x in data]
```
Comment (guewen) [inline features/steps/tools.py]: Not new but could be replaced by `list(data)` directly on the line `result = model(model_name).load(head, list(data), context)`

**Comment by charbeljc** [comment]
Definitively, yes, I will do that :-)

File: features/steps/tools.py (hunk 1 above)
Comment (charbeljc) [inline features/steps/tools.py]: right

**Comment by guewen** [comment]
> Could you push it further and have only one function with arguments ctx, model_name, csvfile, sep=",", options=None,

In fact the separator should  be in `options`

**Comment by charbeljc** [comment]
@guewen, refactored following your suggestions

File: features/steps/tools.py [hunk 2]
```diff
@@ -41,67 +55,99 @@ def impl(ctx):
         cr.close()
 
 
-# TODO REFACTOR and add CSV options support
+def str2bool(s):
+    if s in ('True', 'true'):
+        return True
+    elif s in ('False', 'false'):
+        return False
```
Comment (guewen) [inline features/steps/tools.py]: "1" and "0" are values usually accepted

File: features/steps/tools.py [hunk 3]
```diff
@@ -41,67 +55,99 @@ def impl(ctx):
...
-    values = [x for x in data]
+    values = list(data)
+
     context = ctx.oe_context
+    ctx.loaded_objets = None
+
     if model_name == 'res.users':
         context = dict(context or {}, no_reset_password=True)
+
     if options.get('bulk') in ('True', 'true'):
```
Comment (guewen) [inline features/steps/tools.py]: Should use `str2bool()`

**Comment by guewen** [comment]
Could you wait for reviews before merging?
If you are in a hurry because you need it in a project, you can still modify this option in `common.cfg` and set another branch in `rev`:

```
[sources]
scenarios = git https://github.com/camptocamp/oerpscenario.git egg=false full-path=Scenario/OERPScenario rev=master
```

**Comment by charbeljc** [comment]
Hi @guewen, sorry for this one, I was a bit stressed with my partners.csv not willing to import ...

**Comment by lepistone** [comment]
@charbeljc can you please make another PR addressing the points raised by @guewen?

**Pull request merged**